src/app/shell/builtins/miner.ts

```diff
@@ -0,0 +1,142 @@
+import {Command, IOHandler} from '../command';
+import {ShellApi} from '../shellapi';
+
+export class Miner extends Command {
+  constructor(shellApi: ShellApi) {
+    super('miner', shellApi);
+    this.addDescription('mangages morphcoin miners');
+    this.addSubcommand(MinerLook);
+    this.addSubcommand(MinerWallet);
+    this.addSubcommand(MinerPower);
+    this.addSubcommand(MinerStart);
+  }
+
+  async run(iohandler: IOHandler): Promise<number> {
+    this.showHelp(iohandler.stderr);
+    return 1;
+  }
+}
+
+async function getMinerService(shellApi: ShellApi): Promise<any> {
+  const listData = await shellApi.websocket.msPromise('service', ['list'], {
+    'device_uuid': shellApi.activeDevice['uuid'],
+  });
+  for (const service of listData.services) {
+    if (service.name === 'miner') {
+      return service;
+    }
+  }
+  throw new Error('miner service not found');
+}
+
+class MinerLook extends Command {
+  constructor(shellApi: ShellApi) {
+    super('look', shellApi);
+    this.addDescription('shows your current miner settings');
+  }
+
+  async run(iohandler: IOHandler): Promise<number> {
+    let miner: any;
+    try {
+      miner = await getMinerService(this.shellApi);
+    } catch {
+      iohandler.stderr('Miner service not reachable');
+      return 1;
+    }
+    const data = await this.shellApi.websocket.msPromise('service', ['miner', 'get'], {
+      'service_uuid': miner.uuid,
+    });
+    const wallet = data['wallet'];
+    const power = Math.round(data['power'] * 100);
+    iohandler.stdout('Wallet: ' + wallet);
```

Comment:
You could use Template Strings. (For this and the next few lines)

Reply:
```javascript
const someVar = 'test';

const someString = `This is a ${someVar}.`
```